### PR TITLE
Add functionality to register a webmention comment type

### PR DESCRIPTION
**Pull request by dshanske**
The idea here is a simple registration class for the webmention comment types based on the register_post_type function. 

So far, it incorporates the strings used for the various internal labels and menus, but eventually would add others as needed.

Essentially, it is an alternative to adding strings in five different places when adding a new type. And it will make it easier to add additional configuration options as needed.

If WordPress ever adds its own comment type registration, this would replace that.

### CODE REVIEW

**Comment by dshanske** [comment]
The other thing I want to do with this in future is allow for custom comment handlers to be registered for each type. We had a template request for that in Semantic Linkbacks. It would be nice to accommodate it.

**Comment by pfefferle** [comment]
I thought we wanted to make the current subtypes to be the main comment types? So reply, repost, ... instead of Webmention?

**Comment by dshanske** [comment]
Yes, this is the start of the code for that

**Comment by dshanske** [comment]
So, this code would register reply, like, etc.

File: includes/functions.php [hunk 1]
```diff
@@ -401,27 +437,37 @@ function is_webmention_comment_type( $comment ) {
...
-			break;
-		default:
-			$name = __( 'Response', 'webmention' );
+
+	$name = null;
+
+	if ( is_array( $webmention_comment_types ) && array_key_exists( $type, $webmention_comment_types ) ) {
+		$name = $webmention_comment_types[ $type ]->singular;
 	}
+	if ( ! $name ) {
```
Comment (pfefferle) [inline includes/functions.php:452]: shouldn't this be part of the new class?

**Comment by pfefferle** [comment]
where should the default types be registered? maybe we should also add a class & rel list, to also include https://github.com/pfefferle/wordpress-semantic-linkbacks/blob/master/includes/class-linkbacks-mf2-handler.php#L23

**Comment by dshanske** [comment]
@pfefferle The class and rel list was something I planned for additions. This PR is just the basic design.

**Pull request merged**